### ISSUE TITLE
SOCKET ERROR - RECEIVE error: socket closed

### ISSUE THREAD

**Issue by giosal** [nicehash/NiceHashMiner-Archived#1041]
I get this error on several machines.

I tested on all NH locations (EU, USA, CN, JP, IN, BR) with 1.7.5.12 and 1.7.5.13.
I read one of the issues saying that I need to use different bitcoin addresses for machines on different network.

BUT:
- all my machines are on the same network starting with 188.185
- now I have 173 machines running
- I had to reinstall several machines after windows update bsod'd on them
- before update I had 225 machines running without problem.

Please advise.
If I use different bitcoin adresses, then I'll have to use different stats page for each address, right?

**Comment by giosal** [comment]
Nothing?

**Comment by giosal** [comment]
this seems to be an issue related to network as tracert doesn't finish with success.
any advice on firewall?

**Comment by giosal** [comment]
so, for several days all 223 workers were connected, but now they started to lose connection again.
what's the problem?
@S74nk0

**Comment by giosal** [comment]
only 193 workers have connection now...

**Comment by giosal** [comment]
now only 176 workers have connection, what's happening?

**Comment by giosal** [comment]
@S74nk0 @nicehashdev I also tried to start mining with another Bitcoin address, it still doesn't connect...
so there seems to be a problem with you servers.
I tried on EU, USA, HK, JP, IN and BR

**Comment by smithinoz** [comment]
Hi, yes I have the same issue.  Started 12 hours ago down from 200 to 0.
All mine are on the same network (public IP).
Only 1 will connect, but the web site shows 0 with 0 hash.  No issues with other pools.  Looks like Nicehash suspect a bot attack and shut them down,

**Comment by Genrymoney** [comment]
https://new.nicehash.com/help/repeatedly-getting-socket-error-socket-closed

**Comment by CrizzzSombiii** [comment]
Same  problem.

**Comment by clarity99** [comment]
Same problem and I don't think the explanation on nicehash applies to this case - when it starts happening, it keeps going on all the time, and no mining happens at all. if i change to a different location it works ok for some hours, then again socket closed all the time. Then switch pool and again ok for some time. It can happen from different machines at the same time, with very different CPU configurations (one a VM, one a i5 3.4 ghz desktop). 


[2017-10-19 13:16:44] : Connected. Logging in...
[2017-10-19 13:16:44] : SOCKET ERROR - RECEIVE error:
[2017-10-19 13:16:44] : Pool connection lost. Waiting 10 s before retry.
[2017-10-19 13:16:54] : Connecting to pool cryptonight.eu.nicehash.com:3355 ...
[2017-10-19 13:16:54] : Connected. Logging in...
[2017-10-19 13:16:54] : SOCKET ERROR - RECEIVE error: socket closed
[2017-10-19 13:16:54] : Pool connection lost. Waiting 10 s before retry.
[2017-10-19 13:17:04] : Connecting to pool cryptonight.eu.nicehash.com:3355 ...
[2017-10-19 13:17:04] : Connected. Logging in...
[2017-10-19 13:17:04] : SOCKET ERROR - RECEIVE error: socket closed
[2017-10-19 13:17:04] : Pool connection lost. Waiting 10 s before retry.
[2017-10-19 13:17:14] : Connecting to pool cryptonight.eu.nicehash.com:3355 ...
[2017-10-19 13:17:14] : Connected. Logging in...
[2017-10-19 13:17:14] : SOCKET ERROR - RECEIVE error: socket closed
[2017-10-19 13:17:14] : Pool connection lost. Waiting 10 s before retry.
[2017-10-19 13:17:24] : Connecting to pool cryptonight.eu.nicehash.com:3355 ...
[2017-10-19 13:17:24] : Connected. Logging in...
[2017-10-19 13:17:24] : SOCKET ERROR - RECEIVE error: socket closed
[2017-10-19 13:17:24] : Pool connection lost. Waiting 10 s before retry.
[2017-10-19 13:17:34] : Connecting to pool cryptonight.eu.nicehash.com:3355 ...
[2017-10-19 13:17:34] : Connected. Logging in...
[2017-10-19 13:17:34] : SOCKET ERROR - RECEIVE error: socket closed
[2017-10-19 13:17:34] : Pool connection lost. Waiting 10 s before retry.
[2017-10-19 13:17:44] : Connecting to pool cryptonight.eu.nicehash.com:3355 ...
[2017-10-19 13:17:44] : Connected. Logging in...
[2017-10-19 13:17:44] : SOCKET ERROR - RECEIVE error: socket closed
[2017-10-19 13:17:44] : Pool connection lost. Waiting 10 s before retry.
[2017-10-19 13:17:54] : Connecting to pool cryptonight.eu.nicehash.com:3355 ...

**Comment by qbac** [comment]
I have the same problem:

[2017-10-22 16:19:02] : Connecting to pool cryptonight.eu.nicehash.com:3355 ...
[2017-10-22 16:19:02] : Connected. Logging in...
[2017-10-22 16:19:02] : SOCKET ERROR - RECEIVE error: socket closed
[2017-10-22 16:19:02] : Pool connection lost. Waiting 10 s before retry (attempt
 280).
[2017-10-22 16:19:12] : Connecting to pool cryptonight.eu.nicehash.com:3355 ...
[2017-10-22 16:19:12] : Connected. Logging in...
[2017-10-22 16:19:12] : SOCKET ERROR - RECEIVE error: socket closed
[2017-10-22 16:19:12] : Pool connection lost. Waiting 10 s before retry (attempt
 281).
[2017-10-22 16:19:22] : Connecting to pool cryptonight.eu.nicehash.com:3355 ...
[2017-10-22 16:19:27] : Connected. Logging in...
[2017-10-22 16:19:27] : SOCKET ERROR - RECEIVE error: socket closed
[2017-10-22 16:19:27] : Pool connection lost. Waiting 10 s before retry (attempt
 282).
[2017-10-22 16:19:37] : Connecting to pool cryptonight.eu.nicehash.com:3355 ...
[2017-10-22 16:19:37] : Connected. Logging in...
[2017-10-22 16:19:37] : SOCKET ERROR - RECEIVE error: socket closed
[2017-10-22 16:19:37] : Pool connection lost. Waiting 10 s before retry (attempt
 283).

...
...

**Comment by illtellyoulater** [comment]
I am seeing this right now, anyone? 
Just yesterday I had it working with no issues at all.

Update: it must be a problem at a region-specific NiceHash end, temporarily choosing another region fixes it.

**Comment by mojimi** [comment]
I had this problem with both the Brazilian and American server, switched to European and it worked

**Comment by mojimi** [comment]
Nevermind, now the European is also giving the same problem

**Comment by ericjackson-maconk12ncus** [comment]
This is happening to me also - only with cpu mining. All works well, then socket error. I restart cpu mining then all is well again for some time. Initially the problem was not as frequent, now it is happening all the time and I can't keep cpu mining open. GPU mining is fine.

BTW - I am not getting the retry messages.

**Comment by ericjackson-maconk12ncus** [comment]
This is happening to me also - only with cpu mining. All works well, then socket error. I restart cpu mining then all is well again for some time. Initially the problem was not as frequent, now it is happening all the time and I can't keep cpu mining open. GPU mining is fine.

BTW - I am not getting the retry messages.

**Comment by ericjackson-maconk12ncus** [comment]
This is happening to me also - only with cpu mining. All works well, then socket error. I restart cpu mining then all is well again for some time. Initially the problem was not as frequent, now it is happening all the time and I can't keep cpu mining open. GPU mining is fine.

BTW - I am not getting the retry messages.

**Comment by ericjackson-maconk12ncus** [comment]
This is happening to me also - only with cpu mining. All works well, then socket error. I restart cpu mining then all is well again for some time. Initially the problem was not as frequent, now it is happening all the time and I can't keep cpu mining open. GPU mining is fine.

BTW - I am not getting the retry messages.

**Comment by Cryptosynapse** [comment]
This is happening to me also - only with cpu mining not gpu mining. All works well, then socket error. I restart cpu mining then all is well again for some time. Initially the problem was not as frequent, now it is happening all the time and I can't keep cpu mining open. GPU mining is fine.

BTW - I am not getting the retry messages.

**Comment by mojimi** [comment]
I am using a very weak intel i5 (50 mh/s) just to test NiceHash before buying better geat, is this issue related to having a weak setup? I'm kinda afraid to upgrade now.

**Comment by semtex41** [comment]
Still seeing this problem, it continues to retry every 10 seconds and fails.

**Comment by xermalk** [comment]
This is just silly, everytime i shutdown the miner i have to force a ip change or i wont be able to start mining again.

`[2017-12-03 21:44:00] [INFO] [SOCKET] 12/03/2017 21:44:00|Debug|WebSocket.sendHttpRequest|A request to the server:
                          GET /v2/nhm HTTP/1.1
                          User-Agent: websocket-sharp/1.0
                          Host: api.nicehash.com
                          Upgrade: websocket
                          Connection: Upgrade
                          Sec-WebSocket-Key: +mrggqk0EXwoli4qAyp+Sw==
                          Sec-WebSocket-Version: 13
[2017-12-03 21:44:00] [INFO] [SOCKET] 12/03/2017 21:44:00|Debug|WebSocket.sendHttpRequest|A response to this request:
                          HTTP/1.1 503 Service Temporarily Unavailable
                          Date: Sun, 03 Dec 2017 20:44:02 GMT
                          Content-Type: text/html
                          Transfer-Encoding: chunked
                          Connection: keep-alive
                          Set-Cookie: __cfduid=ddde1ee70ef622ba150705a21007386d61512333842; expires=Mon, 03-Dec-18 20:44:02 GMT; path=/; domain=.nicehash.com; HttpOnly
                          Server: cloudflare-nginx
                          CF-RAY: 3c795c921df08673-ARN
[2017-12-03 21:44:00] [INFO] [SOCKET] 12/03/2017 21:44:00|Fatal|WebSocket.connect|Not a WebSocket handshake response.
[2017-12-03 21:44:00] [INFO] [SOCKET] 12/03/2017 21:44:00|Debug|WebSocket.closeHandshake|Was clean?: False
                            sent: False
                            received: False
[2017-12-03 21:44:00] [INFO] [SOCKET] Connection closed code 1002: An error has occurred while connecting.`

**Comment by AlexVirlan** [comment]
I just checked my workers. I have the same problem since last night. Any idea why?

**Comment by dashony** [comment]
Same issue here with both workers. It works for 1-2 hours or less sometimes and it fails on both machines with the above messages over and over again. It would not resolve on its own until I manually change servers and then I might get another hour or less until it stops working again, then no server would work. I also tried different wallet addresses on different workers and no difference. 
Any luck anyone? (i might need to stick with Kryptex, as it doesn't seem to fail)

**Comment by dseeker** [comment]
same issue. works for a few hours, then stopped working. Restart, same problem. was connecting only to EU, then tried USA, worked again. a few hours passed, now cant connect to any of the servers.
was excited with nicehash, but this is a total dealbreaker

**Comment by dseeker** [comment]
I enabled a VPN connection and servers starting responding.
Maybe an issue with ISPs blocking miners?

**Comment by dashony** [comment]
I’ve tried the VPN and worked for a little while and same thing happened again. Let us know @dseeker if you get a sturdy 48 hours before you get banned. I believe we’re getting a soft ban, that’s why we get disconnecred.

**Comment by dseeker** [comment]
can't say for 48h, cause I shut it down once a day. but it got much better than without the VPN. 
before the error would come within 15/30 minutes and stop all mining. With VPN I get the error but then it's followed by "All pools are dead. Idling..." and a few seconds later it starts again.

So it's definitely not a local issue with the software, either ISP or like you said some kind of server ban
![image](https://user-images.githubusercontent.com/11655138/34437978-5b441bf8-eca3-11e7-8eba-2dc5956656e3.png)

**Comment by dseeker** [comment]
@dashony yeah, after about 12h I started getting the error again without recovery. Changed location of the VPN and it started working again. Damn... Can't be the case that everyone using NiceHash has the same issue. perhaps due to banning slower machines when pool is full?

**Comment by mojimi** [comment]
In the case of cryptonight yes, NiceHash cryptonight pools have a fairly high difficulty so if you're too slow you will get socket timeout all the time. If this is your case just learn to mine Monero directly using XMR-STAK and another pool

If this is not your case, try using NiceHashLegacy which is fairly more updated than this one ironically.

**Comment by J0SH912** [comment]
I was having the same problem, created a new wallet and now it is working fine.

**Comment by dashony** [comment]
@J0SH912 I’ve tried different wallets a few days ago and it worked for a few hours then ran into the same issue. I think @mojimi is right. I only do CPU mining and as many above we seem to have the same issues under the same scenario and we’re getting some soft bans from the servers.

**Comment by emerzon** [comment]
Same issue here. My IPs keep getting banned after some 20/30  minutes running.
I am also doing CPU mining with low hashrate. Is there some fix/workaround?

**Comment by koriakine** [comment]
I have 5 machines with XEON in different IPs networks. 3 of them are working ok, two have the same thing you mentioned. Compiled new XMR-STAK, same error.

**Comment by orphiler** [comment]
same issue over here 👎  help pliz i use server rdp to mine it's CPU

-------------------------------------------------------------------
[2018-01-12 13:30:17] : Start mining: MONERO
[2018-01-12 13:30:17] : Starting double thread, affinity: 0.
[2018-01-12 13:30:17] : hwloc: memory pinned
[2018-01-12 13:30:17] : Starting double thread, affinity: 1.
[2018-01-12 13:30:17] : hwloc: memory pinned
[2018-01-12 13:30:17] : Fast-connecting to cryptonight.hk.nicehash.com:3355 pool ...
[2018-01-12 13:30:17] : Pool cryptonight.hk.nicehash.com:3355 connected. Logging in...
[2018-01-12 13:30:17] : SOCKET ERROR - [cryptonight.hk.nicehash.com:3355] RECEIVE error: socket closed
[2018-01-12 13:30:47] : Fast-connecting to cryptonight.hk.nicehash.com:3355 pool ...
[2018-01-12 13:30:48] : Pool cryptonight.hk.nicehash.com:3355 connected. Logging in...
[2018-01-12 13:30:48] : SOCKET ERROR - [cryptonight.hk.nicehash.com:3355] RECEIVE error: socket closed

**Comment by GalballyJ** [comment]
same problem here but on a smaller scale. all the machines from location1 cannot connect and get sent a RESET after login. machines ni location 2 and 3 work ok tho. same error on all the pools so its not region specific

**Comment by orphiler** [comment]
and what is the solution ?



<https://mailtrack.io/> Sent with Mailtrack
<https://chrome.google.com/webstore/detail/mailtrack-for-gmail-inbox/ndnaehgpjlnokgebbaldlmgkapkpjkkb?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality>

2018-01-12 15:03 GMT+01:00 GalballyJ <notifications@github.com>:

**Comment by pfilias** [comment]
I just started with nicehash today and am getting this issue right from the get-go. Nobody has found a solution yet?

Mine also outputs:

> SOCKET ERROR - SEND error: An established connection was aborted by the software in your host machine.

That's nuts, because I have disabled my Norton firewall.

I did a little more digging with Norton and found this:
> Category: Intrusion Prevention
Date & Time,Risk,Activity,Status,Recommended Action,IPS Alert Name,Default Action,Action Taken,Attacking Computer,Destination Address,Source Address,Traffic Description
1/20/2018 5:18:56 PM,High,An intrusion attempt by cryptonight.usa.nicehash.com was blocked.,Blocked,No Action Required,System Infected: Bitcoinminer Activity 6,No Action Required,No Action Required,"cryptonight.usa.nicehash.com (198.11.220.44, 3355)","DESKTOP-K0P1NLV (192.168.1.100, 52698)",cryptonight.usa.nicehash.com (198.11.220.44),"TCP, Port 3355"
Network traffic from <b>cryptonight.usa.nicehash.com</b> matches the signature of a known attack.  The attack was resulted from \DEVICE\HARDDISKVOLUME8\USERS\____\APPDATA\ROAMING\NHM2\BIN\XMR-STAK-CPU\XMR-STAK-CPU.EXE.  To stop being notified for this type of traffic, in the <b>Actions</b> panel, click <b>Stop Notifying Me</b>.

**Comment by xberg** [comment]
Same problem here. New to Nicehash. 2 workers. Worked fine for 1 week and today I have been blocked. 
Of course no fishy activity on my side.
Sometimes I am able to connect again for a few minutes, then I lose the connection with the server. Tried both CPU and GPU and different algos. 
Disabled any firewall or anti virus. 
Each worker has 2 CPU running + 1 GPU. 
EDIT: lasted 3 hours and then restarted without me changing anything.

**Comment by nicehashdev** [comment]
Please note that if your miner does not send any share in 5 minutes your connection will be temporarily blocked.

The difficulty for mining Cryptonight is very high at the moment because a lot of miners are mining it and miners with better hardware get shares faster than you.
	
Kind regards,
NiceHash Team

**Comment by xberg** [comment]
Here we go again: both my miners stopped at the same time. No more connection.
One connected to US, the other to Europe. All on different algo. 
I'm losing mining time with this problem. 
Is there a way to set an alternative mining pool?

**Comment by thecadman99** [comment]
Same problem- 20 miners work for 24 hours or so, then off for the same amount of time.   Change wallet and still no connection.  Really need help with this!

**Comment by dseeker** [comment]
I'm wondering if something like this helps, has anyone tried?
https://github.com/xmrig/xmrig-proxy

**Comment by AlexVirlan** [comment]
**I have a working solution!** I'm working on an app that can manage all this disconnections and keep the NHM working.
I just need just a little bit more time for some final testing, and I will release it for free.

**Comment by emerzon** [comment]
According to the docs, it cannot be used with Nicehash.

2018-01-30 10:24 GMT+01:00 D See Ker <notifications@github.com>:

**Comment by emerzon** [comment]
Awesome!!! Waiting for it :)

2018-01-30 10:28 GMT+01:00 AAv.Soft <notifications@github.com>:

**Comment by AlexVirlan** [comment]
@emerzon Wait until you see it. It won't affect the connection. It does't even touches the connections or anything else about the NHM's system or protocol.
I'm doing this release to help those people who have the "socket problem".

**Comment by xberg** [comment]
@AAv-Soft That's fantastic news. So what exactly is the root cause of this error?

**Comment by AlexVirlan** [comment]
@xberg Thank you for your interest. As @nicehashdev mentioned, the NH pool is configured in such a way that "**if your miner does not send any share in 5 minutes your connection will be temporarily blocked** ". 
Because as the difficulty for mining increases, miners with better hardware get shares faster.
So you get blocked for not submitting results and for trying to get in every few seconds. 
So this app will check your CPU or GPU status, and if you are not mining for a period of time, it will automatically stop mining, wait until the block time is over, then switch the mining server and start again. And you won't have to do this by hand.

**Comment by xberg** [comment]
@AAv-Soft Somehow I doubt this is the real reason.
Otherwise why would ALL my miners stop at exactly the same time? All using different algos. Do you think that just because 1 worker is inefficient nicehash blocks all? If that's the case they have a real problem.

**Comment by AlexVirlan** [comment]
@xberg 
1. What NHM version are you using?
2. Are you mining on CPU and GPU at the same time or only one of them?
3. Are you mining with all algos on the same BTC address?

**Comment by xberg** [comment]
@AAv-Soft 
1. Nicehash 1.8.1.16 legacy
2. Yes: CPU and GPU at the same time
3. Yes: all algos and all workers on the same address.

**Comment by mojimi** [comment]
@nicehashdev Ok but why not implement different difficulty ports like every cryptonight pool? The whole appeal of cryptonight is that you can mine using your cpu, what's the point if you can only mine with 500h/s+?

**Comment by AlexVirlan** [comment]
@mojimi Exactly my idea. That's the problem. 
@xberg "3. Yes: all algos and all workers on the same address." Well, my theory is that they block the wallet address.

**Comment by xberg** [comment]
@mojimi I am getting blocked with over 2500 h/s in cryptonight so I really don't think this explanation is correct. 
@AAv-Soft If that's the case it's a serious bug on their side. Imagine a mega farm in which someone connects a small laptop to demo mining. The laptop under performs and the whole farm is blocked :)
Or worse: I could connect my laptop to your wallet and block you? Nonsense. 

I think @nicehashdev does not know the root cause and just came up with the first idea he had. Which of course is the natural thing to do, but the issue deserves more exploration.

**Comment by mojimi** [comment]
@xberg I mine 2400 h/s Cryptonight to NiceHash and had no problems.

Have you tried using the legacy miner? Despite the name it is the more updated version.

But here's how I do it; I use xmr-stak miner and point it to NiceHash's Cryptonight pool, I use the BR server as its closer to me, works like a charm.

**Comment by xberg** [comment]
@mojimi Yes: I use the legacy 1.8.1.16.

**Comment by xberg** [comment]
@mojimi : what do you mean by you point it to Nicehash's CN pool? I just define the "service location". Do you precisely define the pool? 
As in:
stratum+tcp://cryptonight.eu.nicehash.com:3355
?

**Comment by mojimi** [comment]
@xberg I meant that I don't use NiceHash's miner, I use XMR-Stak Cryptonight miner as I know it will always be the most profitable in my hardware.

Just set the pool config as "cryptonight.eu.nicehash.com:3355" and enable the NiceHash option, its pretty simple

**Comment by xberg** [comment]
Sometging interesting just happened with EthDcrMiner64:
`ETH: Stratum - connecting to 'daggerhashimoto.eu.nicehash.com' <159.122.29.199> port 3353
ETH: Stratum - Connected (daggerhashimoto.eu.nicehash.com:3353)
GPU0 t=55C fan=13%
Socket was closed remotely (by pool)
ETH: Job timeout, disconnect, retry in 20 sec...`

this lasted about 10 minutes, followed by:
`ETH: Stratum - connecting to 'daggerhashimoto.eu.nicehash.com' <159.122.29.199> port 3353
ETH: Stratum - Connected (daggerhashimoto.eu.nicehash.com:3353)
ETH: Authorized
Setting DAG epoch #7...
Setting DAG epoch #7 for GPU0
GPU0 DAG creation time - 3724 ms
Setting DAG epoch #7 for GPU0 done
GPU0 t=60C fan=20%
ETH: 01/31/18-12:55:20 - New job from daggerhashimoto.eu.nicehash.com:3353 (FAILOVER 2)`

And since then it's been using this failover2.

Has Nicehash changed anything for the better?

**Comment by AlexVirlan** [comment]
@xberg about what you said "Or worse: I could connect my laptop to your wallet and block you? Nonsense.". I do have some theory about the problem, hear me out.

So, you get blocked (on Europe) and you CAN mine on another pool (also known as "server" or "service location" in NHM). But you get blocked on all of the pools eventually and you need to wait for some time (I think that is around 60 minutes).
The big question here is **what information do they use to block a specific machine**.
I think that the BTC Address must be involved so they can track it trough the platform and pools. But they can also do a fingerprint (or unique ID) for every machine (I used this technique for licensing on other projects in the past).
For example: **Some_Hardware_Info + Some_Software_Info + Wallet_Address = The_Fingerprint**
(or even include some network info or the public IP)
Doing this you can solve the "I could connect my laptop to your wallet and block you?" problem, because **they can block a specific connection, on a specific machine, with a specific wallet address**.

I didn't have time to look around in the NHM source code yet, but I will do this as soon as I can.
If everyone saw something, please tell me.

**Comment by xberg** [comment]
Again I was connected to FAILOVER 1 after being disconnected for a few minutes, so everything seems to be working fine now. 
The error I had was:
Socket was closed remotely by pool.

But in a minute or 2 I was connecting to the failover pool. 

I will continue monitoring this but it's looking promising.

**Comment by gagstheminer** [comment]
I guess this problem has not been resolved yet.. I used to get this socket error, but it would go away for a while if I restarted my router / modem. Thinking it is a soft ban from the service provider, I switched to another service provider but the problem persists...

**Comment by gjta** [comment]
Is anyone still having this problem? I am new to nicehash, and meanwhile my gpu#1 is running fine getting 65.273mh/s, my cpu1 with cryptonight is not running at all and im also getting the error msg after trying to connect:

"socket error - send error: an established connection was aborted by the software in your host machine"

Any update or information on how this can be fixed would be much appreciated, many thanks in advance.

**Comment by AlexVirlan** [comment]
@gjta Mine using ONLY your GPU and see what happens. Then post a feedback here if you want.

**Comment by gjta** [comment]
@AAv-Soft okay so I turned off the CPU, and clicked start on NH again.  GPU works fine, and only one window opened once I clicked start which shows "core l algorithm 'equihash' total speed: 678,331729 h/s" etc etc,

Fyi my current CPU is an intel i7 8700k coffee lake 3.7ghz, if that helps.

**Comment by emerzon** [comment]
TL;DR: Don't worry about this error if your hash rate is not super low.

TL;DR2: Slow clients are not supported by nicehash.

Nicehash imposes a 10 min timeout on each connection. For Cryptonight on
CPU you will normally everyday have over 10 mins without any communication
to server due to the high difficulty currently.
When required your client will establish a new connection and life goes on.
The problem is if your hash rate is too low, it means that you will reach
timeout very often, thus, triggering a soft ban on your IP.

Em seg, 12 de mar de 2018 13:29, gjta <notifications@github.com> escreveu:

**Comment by gjta** [comment]
@emerzon sir can you kindly explain that in plain english, this my first day starting this journey, are you saying that my hash rate is too slow? hence the timeout i am getting?

**Comment by AlexVirlan** [comment]
@gjta 
> can you kindly explain that in plain english

The mining difficulty is very high, so CPUs can't mine as fast as GPUs (or ASIC miners). And if you mine on your CPU, you (most likely) won't be able to submit any hashes faster than 10 minutes, so NH will give you a "time out".

Read more about this by "googleing" around.

**Comment by emerzon** [comment]
This i​s specific to Cryptonight. Modern CPU will most of the times be able
to send a share in less than 10 minutes, but not always. So in these cases
you may see a time-out, but your client will eventually recover. Now if
your CPU is too slow and will always send shares in periods greater than 10
minutes, you will get a soft ban.


2018-03-12 13:46 GMT+01:00 AAv.Soft <notifications@github.com>:

**Comment by gjta** [comment]
@AAv-Soft thank you sir for the clarification. Is there a way around this? so it can be resolved and I am able to run this algorithm? 

@emerzon many thanks for that, that makes more sense.

**Comment by gjta** [comment]
@AAv-Soft thank you sir for the clarification. Is there a way around this? so it can be resolved and I am able to run this algorithm?

**Comment by AlexVirlan** [comment]
@gjta If you have a GPU, don't worry about mining Cryptonight with your CPU.
The GPU is much more faster. For now, there's no way around this, since the NH servers are programmed to  time-out the slow miners.
Stay subscribed to this page and you will get notified about possible changes.
